comps/router/src/integrations/controllers/semantic_router_controller/semantic_router_controller.py

```diff
@@ -0,0 +1,84 @@
+# semantic_router_controller.py
+
+import logging
+import os
+from comps.router.src.integrations.controllers.base_controller import BaseController
+from semantic_router.routers import SemanticRouter
+from semantic_router.encoders import OpenAIEncoder, HuggingFaceEncoder
+from semantic_router import Route
+# from decorators import log_latency
+from dotenv import load_dotenv
+
+load_dotenv()
+hf_token = os.getenv("HF_TOKEN", "")
+openai_api_key = os.getenv("OPENAI_API_KEY", "")
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+)
+
+class SemanticRouterController(BaseController):
+    def __init__(self, config, api_key=None, model_map=None):
+        self.config = config
+        self.model_map = model_map or {}
+        self.controller_type = config.get("controller_type")
+        embedding_provider = self.config.get("embedding_provider", "huggingface")
+
+        logging.info(f"Embedding Provider: {embedding_provider}")
+
+        if not api_key:
+            raise ValueError("Please set api_key for semantic router.")
+
+        try:
+            if embedding_provider == "huggingface":
+                self.encoder = HuggingFaceEncoder(
+                    name="BAAI/bge-base-en-v1.5",  # Specify your Hugging Face model here
+                    model_kwargs={"token": hf_token},
+                    tokenizer_kwargs={"token": hf_token}
+                )
+            elif embedding_provider == "openai":
+                os.environ["OPENAI_API_KEY"] = api_key
+                self.encoder = OpenAIEncoder()
+            else:
+                raise ValueError(f"Unsupported encoder type: {embedding_provider}")
+
+        except Exception as e:
+            logging.error(f"Failed to initialize encoder for {embedding_provider}: {e}")
+            raise
+
+        self._build_route_layer()
+
+    def _build_route_layer(self):
+        # Build routes from the local controller config
+        routes = self.config.get("routes", [])
+        route_list = [Route(name=route["name"], utterances=route["utterances"]) for route in routes]
+
+        # Reinitialize SemanticRouter to clear previous embeddings when switching models
+        self.route_layer = SemanticRouter(encoder=self.encoder, routes=route_list)
+        logging.info("[DEBUG] Successfully re-initialized SemanticRouter with fresh embeddings.")
+
+    def route(self, messages):
```

Comment:
good to trace route with telemetry
```suggestion
   @opea_telemetry
    def route(self, messages):
```